hydroflow/src/util/cli.rs

```diff
@@ -14,6 +16,21 @@ async fn accept_incoming_connections(
     bind_results
 }
 
+pub async fn launch_flow(mut flow: Hydroflow) {
+    let stop = tokio::sync::oneshot::channel();
+    tokio::task::spawn_blocking(|| {
+        let mut line = String::new();
+        std::io::stdin().read_line(&mut line).unwrap();
+        assert!(line.starts_with("stop"));
+        stop.0.send(()).unwrap();
+    });
+
+    tokio::select! {
+        _ = stop.1 => {},
+        _ = flow.run_async() => {}
+    }
+}
```

Comment:
Probably worth adding a cancellation token or something to `run_async()` in the future